### PR TITLE
Babel cache

### PR DESCRIPTION
**Pull request by nmilenkovic**
Fix problem with babel cache problem, ability to switch versions with command line args

### CODE REVIEW

File: examples/with-universal-configuration/package.json [hunk 1]
```diff
@@ -3,7 +3,9 @@
   "version": "1.0.0",
   "scripts": {
     "dev": "next",
-    "build": "next build",
+    "clean": "rimraf .\\node_modules\\.cache",
+    "build-dev": "npm run clean && next build --env=development",
```
Comment (sedubois) [inline examples/with-universal-configuration/package.json]: why not simply 
```
"build-dev": "npm run clean && next build",
"build-prod": "npm run clean && NODE_ENV=production next build"
```

?

File: examples/with-universal-configuration/package.json [hunk 2]
```diff
@@ -12,6 +14,8 @@
     "react-dom": "^15.4.2"
   },
   "devDependencies": {
+    "command-line-args": "^4.0.2",
```
Comment (sedubois) [inline examples/with-universal-configuration/package.json]: I didn't understand why this package was needed

File: examples/with-universal-configuration/package.json (hunk 1 above)
Comment (nmilenkovic) [inline examples/with-universal-configuration/package.json]: maybe we need to pass additional parameters like instance (we need to pass 2 params)

Comment (sedubois) [inline examples/with-universal-configuration/package.json]: Then can just add another env var.

IMHO that's unrelated to what the example is trying to convey. There's no need for `command-line-args` to get universal config working.

File: examples/with-universal-configuration/package.json (hunk 2 above)
Comment (nmilenkovic) [inline examples/with-universal-configuration/package.json]: I think that this package makes params from command line easy to collect

**Comment by timneutkens** [comment]
@arunoda the babel-loader caching issue here makes me think we should clear `node_modules/.cache` when there is no `BUILD_ID` present in `.next`, so we are 100% sure that no babel-loader cache from development is leaked. What do you think?

**Comment by arunoda** [comment]
@timneutkens then in development, it won't get used.
What we should do it detect the changes in `.babelrc` and try to clean it and start the process again.

In the deployment we could clean it, but I rather let it to be done by the user.

**Comment by timneutkens** [comment]
@arunoda yeah lets document the way someone has to clear the cache instead of flushing it ourselves.

**Comment by timneutkens** [comment]
Closing this in favor of #1432.

**Pull request closed without merging**